sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py

```diff
@@ -220,7 +220,7 @@ def generate_directory_sas(
     :param permission:
         The permissions associated with the shared access signature. The
         user is restricted to operations allowed by the permissions.
-        Permissions must be ordered read, write, delete, list.
+        Permissions must be ordered read, add, create, write, delete.
```

Comment:
Thanks so much for taking time to find the proper permission we want to expose to user! ^_^
let's make it read, create, write, delete(add is for append blob, in datalake all files are acatually a blockblob)

Reply:
It is typed as BlobSasPermissions, which has read, add, created, write, delete
https://github.com/Azure/azure-sdk-for-python/blob/6f606828aeb49e1259d8d7af5da05a559bb84237/sdk/storage/azure-storage-blob/azure/storage/blob/_models.py#L959

We may need to subclass to account for directory.

Reply:
I found the doc for this file is not very accurate! Thanks for helping me finding this. https://github.com/Azure/azure-sdk-for-python/blob/5f3b14d7260831fdc1404600192fb759e49bc85f/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py#L21-L24 I created a [PBI](https://msazure.visualstudio.com/One/_workitems/edit/5815390) to fix the doc. 

Here is the [DirectorySasPermission ](https://github.com/Azure/azure-sdk-for-python/blob/5f3b14d7260831fdc1404600192fb759e49bc85f/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py#L403) class
https://github.com/Azure/azure-sdk-for-python/blob/5f3b14d7260831fdc1404600192fb759e49bc85f/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_models.py#L403-L422